Sources/NIO/Socket.swift

```diff
@@ -206,13 +206,15 @@ typealias IOVector = iovec
     ///     - storageLen: The size of the storage itself.
     ///     - controlBytes: A region of a buffer into which control data can be written. This parameter will be modified on return to be
     ///         the slice of the data actually written into, if any.
+    ///     - controlMessagesReceived: If data is received this is populated using the received control bytes.
     /// - returns: The `IOResult` which indicates how much data could be received and if the operation returned before all the data could be received
     ///     (because the socket is in non-blocking mode)
     /// - throws: An `IOError` if the operation failed.
     func recvmsg(pointer: UnsafeMutableRawBufferPointer,
                  storage: inout sockaddr_storage,
                  storageLen: inout socklen_t,
-                 controlBytes: inout Slice<UnsafeMutableRawBufferPointer>) throws -> IOResult<Int> {
+                 controlBytes: inout Slice<UnsafeMutableRawBufferPointer>,
+                 controlMessagesReceived: inout UnsafeControlMessageCollection?) throws -> IOResult<Int> {
```

Comment:
So this interface isn't really doing its job anymore: the `controlBytes` were `inout` because the plan had been to return the `controlBytes` and let the parsing happen outside it. That's clearly not happening anymore with this refreshed API, so instead we should consider at the _very least_ dropping `Slice` and `inout` because we don't need that slicing mess.

More broadly it'd be nicer to be able to pass a struct in here that more clearly communicates the behaviour we want (namely that the control message collection is tied to the lifetime of this pointer.

Reply:
Still feels pretty awkward to be honest.